dali/kernels/slice/slice_gpu.cuh

```diff
@@ -126,59 +171,71 @@ class SliceGPU {
            const std::vector<SliceArgs<OutputType, Dims>> &slice_args) {
     const auto num_samples = in.size();
 
-    detail::SliceSampleDesc<Dims>* sample_descs_cpu =
-      context.scratchpad->Allocate<detail::SliceSampleDesc<Dims>>(AllocType::Host, num_samples);
+    OutputType *fill_values_cpu =
+        context.scratchpad->Allocate<OutputType>(AllocType::Host, num_samples * nfill_values_);
+    for (int i = 0; i < in.size(); i++) {
+      if (default_fill_values_) {
+        assert(nfill_values_ == 1);
```

Comment:
User error or DALI issue? Assert or throw?

Reply:
DALI issue. `default_fill_values_` is selected when original `nfill_values_` is 0, then we make it 1. So, at this point we expect nfill_values_ to be one. Otherwise, there is a logic error